libs/wingsdk/src/platform/platform-manager.ts

```diff
@@ -97,19 +104,14 @@ export class PlatformManager {
     });
   }
 
-  // This method is called from preflight.js in order to return an App instance
-  // that can be synthesized
-  public createApp(appProps: AppProps): App {
-    this.createPlatformInstances();
-
-    let appCall = this.platformInstances[0].newApp;
-
-    if (!appCall) {
-      throw new Error(
-        `No newApp method found on platform: ${this.platformPaths[0]} (Hint: The first platform provided must have a newApp method)`
-      );
+  public async storeSecrets(secretNames: string[]): Promise<any> {
+    if (!this.storeSecretsHook) {
+      throw new Error("No createSecrets method found on platform");
```

Comment:
@eladb not sure its that straight forward, originally I wanted to go with that option of just having platform[0] be the expected platform for storing secrets however I imagine that some `tf-aws` users might want to override this if they are using a custom platform that uses secure strings in SSM. 

So the behavior atm is the last platform that contains the `storeSecrets` hook is the one that is used. This way I could do `wing compile -t tf-aws -t my-secrets-platform`

So now to the error message, if someone compiles with `wing compile -t x -t y -t z` and none of them containe a `storeSecrets` hook, which platform do we complain about

In my most recent commits I update this error to be
```
throw new Error(`No storeSecrets method found on any platform`);
```

let me know what you think.

Reply:
I like Elad's suggestion because I think for the average user the mention of `createSecrets` would be confusing. It's also not reasonably actionable since most people won't jump to implementing their own handling of this. That might be an error that shows up with `--verbose`.

As for which platform to mention in the error, would say we could mention all of them (with special handling for the common case of 1 platform)

Reply:
Perhaps the error could be:

```
Cannot find a platform or platform extension that supports storing secrets
```